_posts/2016-09-21-approaching-tdd-outside-android.md

```diff
@@ -0,0 +1,159 @@
+---
+layout: post
+asset-type: post
+name: Approaching Outside-in TDD on Android I
+title: 'Approaching Outside-in TDD on Android I'
+author: Carlos Morera de la Chica
+image:
+    src: /assets/img/custom/blog/outside-in.png
+tags: 
+- android
+- tdd 
+- design
+canonical:
+    name: my personal blog
+    href: https://carlosmchica.github.io/approaching-tdd-outside-android/
+
+---
+
+<a href="http://martinfowler.com/articles/mocksArentStubs.html#DrivingTdd">Outside-in Test-Driven Development (TDD)</a> can be a challenge to implement. In this 3-part post series, we would like to share our experiences applying it to Android development and offer some practical tips for doing so yourself.
+
+As Android developers, we have been trying to apply the inside-out TDD style to our daily workflow but we thought that there was something missing.
+
+In this blog post series, we would like to share our experience applying Outside-in TDD to Android development. In this first post of the series we will introduce the necessary concepts and present our broad approach to the problem.
+
+Bearing in mind that we already have an architecture in our system, in most of the cases, we know the pieces that compose a feature's design upfront. That is why Outside-in TDD suits better our workflow. If you already know the design, outside-in let you go faster than the baby steps followed in the inside-out approach.
+
+We have been practising the outside-in workflow, again and again, and we would like to share how we have adapted the outside-in workflow to our platform, Android. In order to do so and to make it easier to follow, we have chosen the "Bank Kata" that Sandro uses on his screencast. We found it useful to show our approach to achieving the same effect on the Android platform, where the tools are different and it can be a bit tricky.
+
+The problem description is as follows:
+
+Create a simple bank application with the following features:
+
+<ul>
+<li>Deposit into Account</li>
+<li>Withdraw from Account</li>
+<li>Show a bank statement</li>
+</ul>
+
+The original kata provides a class with the following structure:
+
+````
+public class Account {
+    public void deposit(int amount) {
+    }
+
+    public void withdraw(int amount) {
+    }
+
+    public void showStatement() {
+    }
+}
+````
+
+And one constraint: You are not allowed to add any other public method to this class.
+We have now used the outside-in TDD workflow in numerous projects and have adapted some aspects of it to suit developing on Android specifically. To demonstrate this, and to make it easier to follow, this post will follow the "Bank Kata" that Sandro Mancuso uses on his 'Outside-In' screencast. (You don't have to watched this screencast to follow along but it's useful as a primer on the concepts.) In his screencast the problem isn't build for Android, but we found it useful to have it as a focus.
```

Comment:
I also renamed the Account class to AccountService as it was creating a lot of confusion. People thought that Account was an Entity.